### PR TITLE
altitude operations

### PR DESCRIPTION
**Pull request by mgross**
Now it is possible to create altitude/elevation maps (data comes from package elevatr):
grafik

grafik.

Also custom maps are now possible (user must provide longitude (x), latitude (y) and value (x) data; note that each possible x-y combination must exist, i.e. full data matrix):
grafik

Weighting inside LocatoR with said altitude map (or any other map) is possible. 
We have the option to set negative weighting values to 0 and to apply inverse exponential decay weighting. In the latter case, we can apply a halftime. This can make sense when we weight with altitude for example:
 For a half-time of 1000m, locations with 1000 altitude get half the probability, with 2000m 1/4 of the probability and so forth.

Using Lives 13C data with 13C -18/1 as point gives this without weighting:
grafik
However with weighting, higher altitudes are downweighted:
grafik
With negative values set to zero weight oceans and areas below sealevel get 0 weight:
grafik

### CODE REVIEW

**Comment by arunge** [comment]
@mgross Unfortunately, this feature requires a higher version of `elevatr` than we have currently running, that is why the checks fail:
image

Possibly, you need exactly version 0.99.0 of `elevatr`, correct?

I need to discuss with @jan-abel-inwt what way is best for our system to get the higher version into our docker container.
I will update you when we found a solution!

**Comment by mgross** [comment]
@arunge, I can try with 0.4.2

**Comment by mgross** [comment]
@arunge ok, it also works for older version

**Review by arunge — CHANGES_REQUESTED** [review]
- In general, the feature seems to work, I have some minor suggestions.
- However, I tested to create a new _elevation_ map _from scratch_. I did not upload or import any data. After naming the map and pressing _Run_ a progress bar occurs, but nothing seems to happen after that. (The main panel remains empty.)
- Checking "Saved maps" gives a new map, but it looks empty. The thumbnail seems not to be there:
  https://github.com/Pandora-IsoMemo/iso-app/blob/f549072ddfb8bcd9ed7dfc1ac7144c5c41d01c97/R/04-savedMapsModule.R#L43 
  image
- Only after selecting to load the map, the map becomes visible. Could we display the map directly after it was created in the main panel, without having to load the map?
  image
- Additionally, an error occured in some if-clause, could you check and fix this?
  image

If you need some assistance regarding the display of the map, please let me know.

File: R/03-mapDiff.R [hunk 1]
```diff
@@ -545,11 +545,11 @@ mapDiff <- function(input, output, session, savedMaps, fruitsData, config){
...
-    mapName <- trimws(input$saveMapName)
-    if (mapName == "") {
-      alert("Please provide a map name")
-      return()
-    }
+    # mapName <- trimws(input$saveMapName)
+    # if (mapName == "") {
+    #   alert("Please provide a map name")
+    #   return()
+    # }
```
Comment (arunge) [inline R/03-mapDiff.R:552]: Please, just remove the comments if this is not needed in this case.

File: R/03-mapDiff.R [hunk 2]
```diff
@@ -823,28 +823,29 @@ mapDiff <- function(input, output, session, savedMaps, fruitsData, config){
...
+    #   }),
+    #   type = "user",
+    #   name = mapName
+    # )
+    # maps <- savedMaps()
+    # maps[[length(maps) + 1]] <- map
+    # savedMaps(maps)
+    #
+    # alert(paste0("Map '", mapName, "' was saved"))
+    # updateTextInput(session, "saveMapName", value = "")
```
Comment (arunge) [inline R/03-mapDiff.R:849]: See above.

**Review by arunge — APPROVED** [review]
Why do you not just remove the code that is not needed for this case?

**Comment by arunge** [comment]
@mgross I merged some fixes from the beta into this feature branch. It is just open to remove the comments mentioned above. I could also do this and merge afterwards. Just let me know if you need something. :blush:

**Pull request merged**